deno/lib/__tests__/string.test.ts

```diff
@@ -470,6 +470,19 @@ test("datetime parsing", () => {
   ).toThrow();
 });
 
+test("duration", () => {
+  const duration = z.string().duration();
+  expect(duration.isDuration).toEqual(true);
+
+  duration.parse("P3Y6M4DT12H30M5S");
```

Comment:
Maybe add test for some more cases? Seems like the current regex does not handle various time-only periods [examples](regexr.com/7sgkp)